### DISCUSSION TITLE
Integrating an existing Agent class in the Artsdata data model

### DISCUSSION THREAD

**Discussion by fjjulien** [culturecreates/artsdata-data-model#130] [Data Model] [answered]
### Background
As a [recent CAPACOA report](https://capacoa.ca/en/2024/07/a-tipping-point-towards-open-data/) found, many industry platforms have a concept designating an "artist" or "performer" agent that can be either a `schema:Person` or `schema:Organization` (or subclass thereof).

### Specific use case
The Arts Touring Connector (ATC) is a platform designed to facilitate the planning of show tours. This process, known as "block programming", is carried out by presenting organizations and the networks that support them. Two key classes closely associated with the Tour concept in the ATC data model are: `Artist` and `Show`. As a first step in the ATC-Artsdata integration, ATC would like their users to be able to autocomplete Artist names against Artsdata entities.

### Problem
Artsdata does not have an equivalent class to the ATC Artist class. Therefore ATC developers must reconcile Artists twice: once against Person type entities, and once against Organization type entities.

### Solution
As an immediate solution to ATC's need, we would like to integrate into the Artsdata data model an existing class from a formal RDF ontology that would be a union of Person and Organization entities (i.e. an "Agent" class).

### Tasks
We need help from the Artsdata community to:

- [ ] Identify candidate Agent classes;
- [ ] Achieving consensus on which class to import into the Artsdata data model.

### See also
[Declaring a "Performing Artist" class in Artsdata #131](https://github.com/culturecreates/artsdata-data-model/discussions/131)

**Comment by fjjulien** [comment]
Voici une liste non-exhaustive de classes candidates pour le concept _Agent_, dans le sens d'entité capable d'effectuer des actions. Lorsque le libellé et la description sont disponibles en français, ils apparaissent sous la description en anglais.

- **[dcterms:Agent](https://prefix.zazuko.com/dcterms:Agent)** - [Dublin Core Metadata Initiative](https://en.wikipedia.org/wiki/Dublin_Core) ([details](https://lov.linkeddata.es/dataset/lov/vocabs/dcterms))
Comment en: A resource that acts or has the power to act.
- **[foaf:Agent](https://prefix.zazuko.com/foaf:Agent)** - [Friend of a Friend](https://en.wikipedia.org/wiki/FOAF) ([details](https://lov.linkeddata.es/dataset/lov/vocabs/foaf))
Comment en: An agent (eg. person, group, software or physical artifact).
- **[crm:E39_Actor](https://prefix.zazuko.com/crm:E39_Actor)** - [CIDOC Conceptual Reference Model (CRM)](https://en.wikipedia.org/wiki/CIDOC_Conceptual_Reference_Model) ([details](https://lov.linkeddata.es/dataset/lov/vocabs/crm))
Comment en: This class comprises people, either individually or in groups, who have the potential to perform intentional actions of kinds for which someone may be held responsible. 
Libellé fr: Actant
Commentaire: Cette classe comprend les personnes, que ce soit des individus ou des groupes, qui ont le potentiel d’effectuer intentionnellement des actions pour lesquelles elles peuvent être tenues responsables.
- **[dbo:Agent](https://prefix.zazuko.com/dbo:Agent)** - [DBpedia Ontology](https://en.wikipedia.org/wiki/DBpedia) ([details](https://lov.linkeddata.es/dataset/lov/vocabs/dbpedia-owl))
Comment en: Analogous to a foaf:Agent, an agent is an entity that acts. This is intended to be the super class of Person and Organisation.
Libellé fr: Agent
Commentaire: Equivaut à foaf:Agent, un agent est une entité qui agit. C'est supposé être la super classe de Person et de Organisation.
- **[rdac:C10002 "agent"](https://www.rdaregistry.info/Elements/c/?language=fr#C10002)** - [Resource Description and Access Classes](https://en.wikipedia.org/wiki/Resource_Description_and_Access) ([details](https://lov.linkeddata.es/dataset/lov/vocabs/rdac))
Comment en: An entity who is capable of deliberate actions, of being granted rights, and of being held accountable for its actions.
Libellé fr: Agent
Commentaire: Entité qui est à même d’agir de son propre chef, de se voir accorder des droits et d’être tenue pour responsable de ses actions.

**Comment by christianroy** [comment]
Quelques réflexions en vrac...

La classe la plus proche dans le référentiel des métadonnées du spectacles est Contributeur. Les contributeurs ont un type, qui permet de distinguer les personnes physiques des personnes morales (pris au sens large, incluant les organismes, les collectifs, les groupes, etc.).

Ça me semble être proche du concept d'Agent de DBPedia (parce que c'est la superclasse de Person et Organisation), mais moins de celui de Dublic Core ou Foaf (si je comprends bien, ces ontologies incluent des logiciels ou des objets qui font des actions dans la classe en question). Et la notion de "agir" associée à un agent n'est pas très loin de celui de "contribuer" associer aux contributeurs, quoique différente... Par exemple, si on voulait documenter les critiques données à tel ou tel spectacle, l'auteur de la critique serait un agent, mais pas un contributeur. Même chose pour un organisme qui remettrait un prix à un diffuseur, l'organisme serait agent, mais pas contributeur.

**Reply by fjjulien** [reply]
Merci pour tes réflexions, Christian.

J'ai exclu la classe Contributeur des mes classes candidates sur la base que le référentiel n'est pas encore été publié comme une ontologie RDF (et donc qu'il n'existe pas d'URI pour le concept). Ceci dit, les précisions que tu apportes sur la classe Contributeur sont utiles pour ma compréhension du référentiel.

**Reply by fjjulien** [reply]
Je suis par ailleurs d'accord avec ton analyse que DBpedia est l'ontologie la plus proche du modèle d'Artsdata.

**Comment by fjjulien** [comment]
Puisque l'objectif est d'identifier un concept qui sera une super-classe de [schema:Person](https://schema.org/Person) et [schema:Organization](https://schema.org/Organization) dans Artsdata, nous devrions sans doute choisir une classe qui possède une hiérarchie similaire à l'intérieur de son ontologie.

Trois des classes identifiées plus haut ont des sous-classes similaires : 

- [foaf:Agent](http://xmlns.com/foaf/spec/#term_Agent)
    - superClassOf:
        - [Group](http://xmlns.com/foaf/spec/#term_Group) 
        - [Person](http://xmlns.com/foaf/spec/#term_Person) 
        - [Organization](http://xmlns.com/foaf/spec/#term_Organization)
- [dbo:Agent](https://dbpedia.org/ontology/Agent)
    - superClassOf:
        - [dbo:Person](http://dbpedia.org/ontology/Person)
        - [dbo:Organization](https://dbpedia.org/ontology/Organisation)
- [crm:E39_Actor](http://www.cidoc-crm.org/cidoc-crm/E39_Actor) 
    - superClassOf:
        - [E21 Person](https://cidoc-crm.org/html/cidoc_crm_v7.1.3.html#E21)
        - [E74 Group](https://cidoc-crm.org/html/cidoc_crm_v7.1.3.html#E74)

**Comment by fjjulien** [comment]
While CIDOC-CRM is an interesting ontology from the cultural heritage domain, the [Linked Open Data Vocabularies website](https://lov.linkeddata.es/dataset/lov/vocabs/crm) indicates it has few connections with other ontologies. I reached out to the Canadian Heritage Information Network to (in-)validate this assumption. Here is the response I received from Trang Dang:

> I am not sure about the formal alignment from CIDOC CRM to other schema, but there are a couple of resources that define mapping of data from various projects and schema to CIDOC CRM.
> - A [Mapping page on CIDOC CRM official website](https://www.cidoc-crm.org/report_mappings_res) that lists the mapping of various projects and schema to CIDOC CRM.
> - A [mapping from Linked.Art to Schema.org](https://linked.art/cookbook/mappings/schema/) (and Linked.Art uses CIDOC CRM ontologies)
> - A [metadata standards crosswalk](https://www.getty.edu/research/publications/electronic_publications/intrometadata/crosswalks.html) published by the Getty (CCO, CIDOC CRM, Linked.Art, LIDO, VRA, DC, etc.).

The Linked.Art's mapping to Schema is not a formal mapping, but it does establish useful correspondences, including these two : 
- [E21 Person](https://cidoc-crm.org/html/cidoc_crm_v7.1.3.html#E21) --> [schema:Person](https://schema.org/Person)
- [E74 Group](https://cidoc-crm.org/html/cidoc_crm_v7.1.3.html#E74) --> [schema:Organization](https://schema.org/Organization)

**Comment by fjjulien** [comment, marked as answer]
Après analyse des options et en tenant compte des commentaires de @saumier et @christianroy, nous allons intégrer la classe [dbo:Agent](https://dbpedia.org/ontology/Agent) en vue de répondre à notre _use case_ immédiat.

Nous pourrons reconsidérer l'intégration ou l'aligment avec CIDOC-CRM à une date ultérieure.

**Reply by saumier** [reply]
@fjjulien Artsdata already imports a dbpedia schema that defines dbo:Agent. 

Here is a link to their definition of [dbo:Agent loaded into Artsdata](https://kg.artsdata.ca/entity?uri=http%3A%2F%2Fdbpedia.org%2Fontology%2FAgent) from the following imported file DBpedia Schema 2016-10 (T-BOX). This file used to be available as a download but that link in our documentation no longer works. However you can click the "dereference" link in Nebula to get an authoritative definition from dbpedia. You will see that the definition has not changed, and that [dbo:Person](https://kg.artsdata.ca/entity?uri=http%3A%2F%2Fdbpedia.org%2Fontology%2FPerson) is a sub-class of dbo:Agent as is [dbo:Organization](https://kg.artsdata.ca/entity?uri=http%3A%2F%2Fdbpedia.org%2Fontology%2FOrganisation). 

So using inferencing you can already query Artsdata SPARQL endpoint to search across schema:Person and schema:Organization. 

```
PREFIX schema: <http://schema.org/>
PREFIX dbo: <http://dbpedia.org/ontology/>
select * where { 
     ?s a dbo:Agent ; schema:name ?name .
    graph ?g { ?s a ?type .}
} limit 100 
```

The work that remains is to add this class dbo:Agent to the Artsdata Reconciliation service.

**Reply by saumier** [reply]
Track this issue https://github.com/culturecreates/artsdata-api/issues/17

**Reply by saumier** [reply]
@fjjulien This is deployed and can be used by ATC, IWTS or others.

**Comment by fjjulien** [comment]
I recently realized that dbo:Agent is not stated to be superclass to dbo:Person. This is a significant ontological problem for Artsdata's formalism.

I created an [issue in the dbpedia GitHub](https://github.com/dbpedia/ontology-tracker/issues/41) to understand why this subClassOf relationship isn't stated in Dbpedia.
